### PR TITLE
DOM components

### PR DESCRIPTION
**Pull request by MartynasZilinskas**

### CODE REVIEW

File: packages/simplr-forms-dom/src/components/email.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        const newValue = this.FormStore.GetField(this.FieldId).Value;
+
+        if (this.props.onChange != null) {
+            this.props.onChange(event, newValue, this.FieldId, this.FormId);
+        }
+
+        // TODO: FormProps.OnFieldChange
+    }
+
+    protected get RawDefaultValue() {
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms-dom/src/components/email.tsx]: return type

File: packages/simplr-forms-dom/src/components/email.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * Override the differences between extended interfaces.
+ */
+export interface EmailProps extends DomFieldProps, React.HTMLProps<HTMLInputElement> {
+    name: string;
+    onFocus?: React.EventHandler<React.FocusEvent<HTMLInputElement>>;
+    onBlur?: React.EventHandler<React.FocusEvent<HTMLInputElement>>;
+    onChange?: EmailOnChangeCallback & FieldOnChangeInternalCallback;
+    ref?: any;
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms-dom/src/components/email.tsx]: ref any?

File: packages/simplr-forms-dom/src/components/number.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * Override the differences between extended interfaces.
+ */
+export interface NumberProps extends DomFieldProps, React.HTMLProps<HTMLInputElement> {
+    name: string;
+    onFocus?: React.EventHandler<React.FocusEvent<HTMLInputElement>>;
+    onBlur?: React.EventHandler<React.FocusEvent<HTMLInputElement>>;
+    onChange?: NumberOnChangeCallback & FieldOnChangeInternalCallback;
+    ref?: any;
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms-dom/src/components/number.tsx]: ref any?

File: packages/simplr-forms-dom/src/components/number.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        const newValue = this.FormStore.GetField(this.FieldId).Value;
+
+        if (this.props.onChange != null) {
+            this.props.onChange(event, newValue, this.FieldId, this.FormId);
+        }
+
+        // TODO: FormProps.OnFieldChange
+    }
+
+    protected get RawDefaultValue() {
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms-dom/src/components/number.tsx]: return type

File: packages/simplr-forms-dom/src/components/number.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        return event.currentTarget.value;
+    }
+
+    protected OnChangeHandler: React.FormEventHandler<HTMLInputElement> = (event) => {
+        this.OnValueChange(this.GetValueFromEvent(event));
+
+        const newValue = this.FormStore.GetField(this.FieldId).Value;
+
+        if (this.props.onChange != null) {
+            this.props.onChange(event, newValue, this.FieldId, this.FormId);
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms-dom/src/components/number.tsx]: missing `event.persist`

File: packages/simplr-forms-dom/src/components/email.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        return event.currentTarget.value;
+    }
+
+    protected OnChangeHandler: React.FormEventHandler<HTMLInputElement> = (event) => {
+        this.OnValueChange(this.GetValueFromEvent(event));
+
+        const newValue = this.FormStore.GetField(this.FieldId).Value;
+
+        if (this.props.onChange != null) {
+            this.props.onChange(event, newValue, this.FieldId, this.FormId);
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms-dom/src/components/email.tsx]: missing event.persist

File: packages/simplr-forms/src/modifiers/numeric-to-string.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
+import { FieldValue } from "../contracts/field";
+import { ValueOfType } from "../utils/value-helpers";
+import { BaseModifier } from "./base-modifier";
+
+export interface NumericToStringModifierProps {
+    // delimiter?: string | string[];
+    defaultValue?: any;
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms/src/modifiers/numeric-to-string.ts]: Any?

File: packages/simplr-forms-dom/src/components/email.tsx (hunk 6 above)
Comment (MartynasZilinskas) [inline packages/simplr-forms-dom/src/components/email.tsx]: Fixed.

File: packages/simplr-forms-dom/src/components/number.tsx (hunk 3 above)
Comment (MartynasZilinskas) [inline packages/simplr-forms-dom/src/components/number.tsx]: Fixed.

File: packages/simplr-forms-dom/src/components/number.tsx (hunk 5 above)
Comment (MartynasZilinskas) [inline packages/simplr-forms-dom/src/components/number.tsx]: Fixed.

File: packages/simplr-forms-dom/src/components/select.tsx [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+    renderField(): JSX.Element | null {
+        return <select
+            name={this.FieldId}
+            value={this.Value}
+            onChange={this.OnChangeHandler}
+            disabled={this.Disabled}
+            onFocus={this.OnFocus}
+            onBlur={this.OnBlur}
+
+
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms-dom/src/components/select.tsx]: Multiple new lines

Comment (MartynasZilinskas) [inline packages/simplr-forms-dom/src/components/select.tsx]: Fixed.

File: packages/simplr-forms/src/utils/value-helpers.ts [hunk 9]
```diff
@@ -5,31 +5,61 @@ import { Modifier, Normalizer } from "../contracts/value";
...
 export function ProcessValue<TProcessor, TProcessedResult>(
     components: Array<JSX.Element>,
+    defaultProcessors: JSX.Element[],
     value: FieldValue,
     processorTypeFunctionName: string,
     process: (processor: TProcessor, value: FieldValue) => TProcessedResult): TProcessedResult {
-    if (components == null || components.length === 0) {
-        return value;
-    }
+    // TODO: Remake this.
```
Comment (DovydasNavickas) [inline packages/simplr-forms/src/utils/value-helpers.ts]: What?

Comment (MartynasZilinskas) [inline packages/simplr-forms/src/utils/value-helpers.ts]: Fixed.

File: packages/simplr-forms-dom/src/abstractions/base-dom-field.tsx [hunk 10]
```diff
@@ -44,6 +44,28 @@ export abstract class BaseDomField<TProps extends DomFieldProps, TState extends
         }
     }
 
+    protected GetHTMLProps(props: TProps) {
```
Comment (MartynasZilinskas) [inline packages/simplr-forms-dom/src/abstractions/base-dom-field.tsx]: Fixed.

File: packages/simplr-forms/src/modifiers/numeric-to-string.ts (hunk 7 above)
Comment (DovydasNavickas) [inline packages/simplr-forms/src/modifiers/numeric-to-string.ts]: `string | number`

File: packages/simplr-forms-dom/src/components/select.tsx [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+    RenderValue: SelectValue;
+}
+
+export class Select extends BaseDomField<SelectProps, SelectState> {
+    protected get RawInitialValue(): SelectValue | undefined {
+        if (this.props.multiple ||
+            this.props.initialValue != null) {
+            return this.props.initialValue;
+        }
+        // If select is not multiple, we need to get first option value.
```
Comment (DovydasNavickas) [inline packages/simplr-forms-dom/src/components/select.tsx]: If select does not have multiple options, then we need to get the first option value.

Comment (MartynasZilinskas) [inline packages/simplr-forms-dom/src/components/select.tsx]: Fixed.

**Pull request merged**